Jenkinsfile

```diff
@@ -102,17 +102,64 @@ pipeline {
       }
     }
 
-    stage('Push Image') {
-      when{
-        expression {
-          return env.GIT_BRANCH == 'origin/main';
+    stage('Continuous Deployment') {
+      when {
+        allOf {
+          anyOf {
+            changeset "docker/**"
+            changeset "docs/**"
+            changeset "pom.xml"
+            changeset "src/main/**"
```

Comment:
@ThorodanBrom a newer image will only be pushed to ghcr if changes are detected in any of these files. Anything I should add/remove here?

Reply:
-I think this covers all possibilities. One question I have is would we see the `docker-compose-*` files in the root directory change often? If yes, would it be better to put them in the `docker/` directory?
- The only dir omitted here is the `configs/` - but I guess that a change in the config structure will always be accompanied with a change in the code.

Reply:
- No, there won't be any frequent changes to the docker-compose files. And anyway, a change in docker-compose does not require a newer image or even updating the deployment. The docker swarm deployment is done with [these files](https://github.com/datakaveri/iudx-deployment/tree/master/Docker-Swarm-deployment/single-node/auth) 
- Yes, assuming a change in the config would ideally be always accompanied by a change in code, we don't require explicitly watching for changes there.